src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -132,11 +133,10 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
                 unnecessaryImports = MergeImports(unnecessaryImports);
 
                 EnsureClassificationIdDescriptors();
-                var fadeOut = workspaceOptions.Services.Workspace.Options.GetOption(FadingOptions.FadeOutUnusedImports, language);
+                var fadeOut = ShouldFade(context.Options, tree, language, cancellationToken);
 
                 DiagnosticDescriptor descriptor;
-                var syntaxFacts = workspaceOptions.Services.GetLanguageServices(language).GetRequiredService<ISyntaxFactsService>();
```

Comment:
can we not reference the individual ISyntaxFactsService impls from this layer?

Reply:
No, not from analyzers, as they need to run from command line. See https://github.com/dotnet/roslyn/pull/41363#discussion_r374326929 for a possibility how we could restructure the code to be accessible from analyzers, if needed. IMO, given that language services are mostly used by our code fixes, and they should still be able to access these in normal fashion, I think we can wait to see if our analyzers really need any language service.

Reply:
@CyrusNajmabadi I am going to retain this for now, and create a separate follow up PR to revert the `IsRegularCommentOrDocComment` abstract method added here and switch back to using `ISyntaxFactsService` once available in this layer.